lua/precognition/ffi.lua

```diff
@@ -0,0 +1,29 @@
+local M = {}
+
+---@return ffi.namespace*
+function M.load()
+    if not _G.precog_C then
```

Comment:
Do we need to make this global? Imo we should just import the ffi module where needed.

Reply:
Again I borrowed the implementation from Noice.

Folke has a note there that

```
-- Put in a global var to make sure we dont reload
  -- when plugin reloaders do their thing
```

Reply:
Ahh ok good point!